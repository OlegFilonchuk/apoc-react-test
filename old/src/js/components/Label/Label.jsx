import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Label.less';

function Label(props) {
  const labelClasses = classNames('sc-label', props.className, {
    'sc-required': props.required
  });

  const { children, ...rest } = props;

  return (
    <label {...rest} className={labelClasses}>
      {children}
    </label>
  );
}

Label.propTypes = {
  /**
   * If input should be required. It will add asterisk symbol next to label
   */
  required: PropTypes.bool,

  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * Text value of component and possible elements which `<label>` tag will wrap
   */
  children: PropTypes.node.isRequired
};

Label.defaultProps = {
  children: '',
  className: '',
  required: false
};

export default Label;
