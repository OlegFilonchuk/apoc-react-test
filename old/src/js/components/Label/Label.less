@import '~chameleon/less/variables.less';

.sc-form.sc-horizontal .sc-label {
  padding-right: @padding-sm-vertical;
}
