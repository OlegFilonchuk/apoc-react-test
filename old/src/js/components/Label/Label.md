## Basic label

Simple usage of the **Label** component:

    (
      <div>
        <Label>Simple value</Label>
      </div>
    )

## Required field label
Example of using required label along with input

    (
      <form className="sc-form sc-horizontal">
        <div className="sc-group">
          <Label className="sc-col-sm-4" required>I am required</Label>
          <Input className="sc-col-sm-7" />
        </div>
      </form>
    )

## Auto spreading native React component properties over component

You can add ie. `onClick` to component, even if it's not listed in props

    const callbackFn = () => {
      console.log('Label clicked');
    };

    (
      <div>
        <Label onClick={callbackFn}>Clickable label</Label>
      </div>
    )


## Label with children elements

You can pass elements as `<label>` children, so they will be wrapped by label tag.

    const callbackFn = () => {
      console.log('Label clicked');
    };

    (
      <div>
        <Label htmlFor="simple-button">
          Label for button
          <Button onClick={callbackFn} id="simple-button">Simple button</Button>
        </Label>
      </div>
    )
