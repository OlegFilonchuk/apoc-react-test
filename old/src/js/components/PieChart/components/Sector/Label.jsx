import React from 'react';
import PropTypes from 'prop-types';
import { getDataTestElementProps } from '../../../../utils/dataTestElementPropUtils';

const DEFAULT_FONT_COLOR = 'white';

function Label({
  percentage,
  labelOffsetData,
  isLabelInsideSector,
  offsets: { offsetX, offsetY },
  onRemove,
  id,
  fontSize,
  processedPercentages,
  name,
  coords,
  side,
  isWideEnough,
  nameShouldBeHidden,
  ...restProps
}) {
  const { x, y } = coords;

  const getTextAnchor = () => {
    if (isLabelInsideSector || !labelOffsetData.useBendPointer) {
      return 'middle';
    } else if (side === 'right') {
      return 'start';
    }

    return 'end';
  };

  const fontColor = isLabelInsideSector ? DEFAULT_FONT_COLOR : labelOffsetData.outerLabelColor;

  const verticalOffset =
    !isLabelInsideSector && isWideEnough && !nameShouldBeHidden && labelOffsetData.useBendPointer ? 0 : offsetY;

  const bc = !isLabelInsideSector && labelOffsetData.useBendPointer ? fontSize * 0.4 : 0;

  const correctedY = y + verticalOffset + bc;

  return (
    <text
      x={x + offsetX}
      y={correctedY}
      fontSize={fontSize}
      fill={fontColor}
      textAnchor={getTextAnchor()}
      onClick={() => onRemove(id, percentage * 100)}
      {...getDataTestElementProps(restProps)}
    >
      {name}
      {processedPercentages}
    </text>
  );
}

Label.propTypes = {
  /**
   * Percentage for Label to display
   */
  percentage: PropTypes.number,

  /**
   * Text for Label
   */
  processedPercentages: PropTypes.string,

  /**
   * FontSize for Label
   */
  fontSize: PropTypes.number,
  name: PropTypes.string,
  onRemove: PropTypes.func,
  id: PropTypes.string,
  labelOffsetData: PropTypes.shape({
    labelOffset: PropTypes.shape({
      innerLabelOffset: PropTypes.number,
      outerLabelOffset: PropTypes.number
    }),
    maxSectorAngle: PropTypes.number.isRequired,
    outerLabelColor: PropTypes.string.isRequired
  }).isRequired,
  isLabelInsideSector: PropTypes.bool.isRequired,
  coords: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number
  }).isRequired,
  offsets: PropTypes.shape({
    offsetX: PropTypes.number,
    offsetY: PropTypes.number
  }).isRequired,
  side: PropTypes.string.isRequired,
  isWideEnough: PropTypes.bool.isRequired,
  nameShouldBeHidden: PropTypes.bool.isRequired
};

Label.defaultProps = {
  percentageShift: 0,
  percentage: 0,
  processedPercentages: '',
  fontSize: 16,
  name: '',
  offsetY: 0,
  isRemoveButtonComponent: false,
  onRemove: () => true,
  id: ''
};

export default Label;
