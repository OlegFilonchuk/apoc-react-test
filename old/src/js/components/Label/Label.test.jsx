import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Label from './Label';

const mountLabel = (props = {}, mountMethod = shallow) => mountMethod(<Label {...props}>Text</Label>);

describe('<Label>', () => {
  it('should render <Label>', () => {
    expect(toJson(mountLabel())).toMatchSnapshot();
  });

  it('should render required <Label>', () => {
    expect(toJson(mountLabel({ required: true }))).toMatchSnapshot();
  });

  it('should fire onClick event when clicking on label', () => {
    const onClick = jest.fn();
    const label = mountLabel({ onClick }, mount);

    label.find('label').simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
