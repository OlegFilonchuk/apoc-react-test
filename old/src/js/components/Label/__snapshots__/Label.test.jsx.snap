// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label> should render <Label> 1`] = `
<label
  className="sc-label"
  required={false}
>
  Text
</label>
`;

exports[`<Label> should render required <Label> 1`] = `
<label
  className="sc-label sc-required"
  required={true}
>
  Text
</label>
`;
